modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/PeerSelect.scala

```diff
@@ -95,14 +103,28 @@ object MajorityPeerSelect {
       )
 
     def getPeerSublist(peers: Set[Peer]): F[List[Peer]] = {
-      val sampleSize = (peers.size * peerSampleRatio).toInt
+      val sampleSize = Math.max((peers.size * peerSampleRatio).toInt, minSampleSize)
 
-      Random[F]
-        .shuffleList(peers.toList)
-        .map(_.take(Math.max(sampleSize, minSampleSize)))
+      for {
+        scores <- getTrustScores.map(_.scores)
+        refinedScores = scores.view
+          .mapValues(score => refineV[TrustValueRefinement](score))
+          .collect {
+            case (key, Right(s)) =>
+              key -> s
+          }
+          .toMap
+        candidates = peers.map { p =>
+          p -> refinedScores.getOrElse(p.id, defaultPeerTrustScore)
+        }.toMap
+        size <- MonadThrow[F].fromEither(
+          refineV[Positive](sampleSize).leftMap(new IllegalStateException(_))
+        )
+        samples <- WeightedProspect.sample(candidates, size)
+      } yield samples
     }
 
-    def getSnapshotHashByPeer(peer: Peer, ordinal: SnapshotOrdinal): F[Option[(Peer, Hash)]] =
+    private def getSnapshotHashByPeer(peer: Peer, ordinal: SnapshotOrdinal): F[Option[(Peer, Hash)]] =
```

Comment:
making it private changes nothing as you implement the `PeerSelect`, so only fns declared in `PeerSelect` trait would be exposed

Reply:
Removed the private keywords.